datalad/interface/rerun.py

```diff
@@ -191,8 +191,9 @@ def __call__(
                 'run',
                 ds=ds,
                 status='impossible',
-                message=('unsaved modifications present, '
-                         'cannot detect changes by command'))
+                message=(
+                    'clean dataset required to detect changes from command; '
+                    'use `datalad status` to inspect unsaved changes'))
```

Comment:
FWIW, in real life scenarios I will not be able to use `datalad status` in such cases since takes too long.  Most likely will run `git status`, and that is why (since this information was already collected in `dirty`) it is kinda wasted cycles.  But I guess we could go one step at a time

Reply:
> Most likely will run `git status`, and that is why (since this information was already collected in `dirty`) it is kinda wasted cycles.

My reservations about adding "N-first modified files" to the status (as suggested [here]):

  * With the .dirty interface, we are committing to a simple no parameters, boolean value interface.  If we find a better method to get the same answer (e.g., no need to consume all of status's output), nothing prevents us from changing to it.

  * In terms of saving cycles, the expensive 'git status' calls will be ones with lots of output.  If you're only showing the first N items, there will be a need for a follow-up status call anyway.

[here]: https://github.com/datalad/datalad/issues/3342#issuecomment-497083665

> But I guess we could go one step at a time
>
> I vote to merge and proceed from there

OK, will merge.  Thanks for the review.